docs/deploy.py

```diff
@@ -32,38 +31,42 @@ def ensure_not_exists(path):
 
 
 def main():
+    old_dir = os.getcwd()
     print("Moving to %s." % HERE)
-    with path(HERE):
-        print("Building docs with 'make html'")
-        check_call(['make', 'html'])
-
-        print("Clearing temp location '%s'" % TEMP_LOCATION)
-        rmtree(TEMP_LOCATION, ignore_errors=True)
-
-        with removing(TEMP_LOCATION):
-            print("Copying built files to temp location.")
-            move('build/html', TEMP_LOCATION)
-
-            print("Moving to '%s'" % ZIPLINE_ROOT)
-            os.chdir(ZIPLINE_ROOT)
-
-            print("Checking out gh-pages branch.")
-            check_call(
-                [
-                    'git', 'branch', '-f',
-                    '--track', 'gh-pages', 'origin/gh-pages'
-                ]
-            )
-            check_call(['git', 'checkout', 'gh-pages'])
-            check_call(['git', 'reset', '--hard', 'origin/gh-pages'])
-
-            print("Copying built files:")
-            for file_ in glob(TEMP_LOCATION_GLOB):
-                base = basename(file_)
-
-                print("%s -> %s" % (file_, base))
-                ensure_not_exists(base)
-                move(file_, '.')
+    os.chdir(HERE)
+
+    print("Building docs with 'make html'")
+    check_call(['make', 'html'])
+
+    print("Clearing temp location '%s'" % TEMP_LOCATION)
+    rmtree(TEMP_LOCATION, ignore_errors=True)
+
+    with removing(TEMP_LOCATION):
+        print("Copying built files to temp location.")
+        move('build/html', TEMP_LOCATION)
+
+        print("Moving to '%s'" % ZIPLINE_ROOT)
+        os.chdir(ZIPLINE_ROOT)
+
+        print("Checking out gh-pages branch.")
+        check_call(
+            [
+                'git', 'branch', '-f',
+                '--track', 'gh-pages', 'origin/gh-pages'
+            ]
+        )
+        check_call(['git', 'checkout', 'gh-pages'])
+        check_call(['git', 'reset', '--hard', 'origin/gh-pages'])
+
+        print("Copying built files:")
+        for file_ in glob(TEMP_LOCATION_GLOB):
+            base = basename(file_)
+
+            print("%s -> %s" % (file_, base))
+            ensure_not_exists(base)
+            move(file_, '.')
+
+    os.chdir(old_dir)
```

Comment:
Hmm I'm not sure what you mean by that. Could you elaborate?

I know that using `with` is equivalent to

```
VAR = EXPR
VAR.__enter__()
try:
    BLOCK
finally:
    VAR.__exit__()
```

but in this context, I'm not sure what you mean about using a `finally` block